src/main/scala/pureconfig/ConfigFieldMapping.scala

```diff
@@ -0,0 +1,24 @@
+package pureconfig
+
+trait ConfigFieldMapping[T] {
```

Comment:
I may have to do some tests before we merge this because this gives a lot of flexibility that maybe is not required. I understand that one could use different styles for different `T`s within the same file, hypothetically, but in reality how many times this will happen? If we decide for a more simple system where keys of all types are treated in the same way, then the interface would be much easier and accessible. We would just need a function `String => String` with default `identity` and no `Wrapper`.